test/upload_test.js

```diff
@@ -86,6 +86,31 @@ describe('S3 Webpack Upload', function() {
         .then(() => testHelpers.fetch(`${testHelpers.S3_URL}${BASE_PATH}/${randomFile.fileName}`))
         .then(fileBody => assert.match(fileBody, testHelpers.S3_ERROR_REGEX, 'random file exists'))
     })
+
+    describe('with priority', function() {
+      it('uploads build to s3 in priority order', function() {
+        testHelpers.createOutputPath()
+        const s3Config = {priority: [/css/]}
+        const config = testHelpers.createWebpackConfig({s3Config})
+
+        return testHelpers.runWebpackConfig({config})
+          .then(testForErrorsOrGetFileNames)
+          .then((files) => {
+            const isCss = file => file.endsWith('css')
+            const isJs = file => file.endsWith('js')
+            const cssFileName = files.find(isCss)
+            const jsFilename = files.find(isJs)
+
+            return Promise.all([
+              testHelpers.getS3Object(cssFileName),
+              testHelpers.getS3Object(jsFilename)
+            ])
+          })
+          .then(([cssObject, jsObject]) => {
+            assert.isTrue(cssObject.LastModified.getTime() >= jsObject.LastModified.getTime())
```

Comment:
Note that S3 accuracy is 1 second so it's not possible to test that the desired object _actually_ is uploaded later: Files are likely te be uploaded at the same second. This test is to ensure that it's at leat not uploaded before. If you want to see the real behaviour, I'd suggest to add logging to e.g in uploadPriorityChunk method.